server/machines/kubernetes/helpers.go

```diff
@@ -32,13 +32,19 @@ func GenerateClientSetAction(k8sContext *models.K8sContext, eventBuilder *events
 }
 
 func GetMachineCtx(machinectx interface{}, eb *events.EventBuilder) (*MachineCtx, error) {
-	machineCtx, ok := machinectx.(*MachineCtx)
-	if !ok {
-		err := machines.ErrAssertMachineCtx(fmt.Errorf("asserting of context %v failed", machinectx))
+	machineCtx, err := GetMachineCtxPlain(machinectx)
+	if err != nil {
 		eb.WithSeverity(events.Error).WithMetadata(map[string]interface{}{
 			"error": err,
 		})
-		return nil, err
+	}
+	return machineCtx, err
+}
+
+func GetMachineCtxPlain(machinectx interface{}) (*MachineCtx, error) {
```

Comment:
the eventbuilder in the GetMachineCtx is a pointer , so might just add some nil checks there and pass nil where we dont have access to the eventbuilder . ( will prevent duplication of logic a bit )